diff -ru tmp/hostapd-2.1/src/ap/wpa_auth.c hostapd-2.1/src/ap/wpa_auth.c
--- tmp/hostapd-2.1/src/ap/wpa_auth.c	2017-03-30 07:51:51.459214482 -0700
+++ hostapd-2.1/src/ap/wpa_auth.c	2017-03-30 07:52:35.391215001 -0700
@@ -45,7 +45,7 @@
 
 static const u32 dot11RSNAConfigGroupUpdateCount = 4;
 static const u32 dot11RSNAConfigPairwiseUpdateCount = 4;
-static const u32 eapol_key_timeout_first = 100; /* ms */
+static const u32 eapol_key_timeout_first = 300; /* ms */
 static const u32 eapol_key_timeout_subseq = 1000; /* ms */
 static const u32 eapol_key_timeout_first_group = 500; /* ms */
 
